### DISCUSSION TITLE
Setting Proxies at execution time

### DISCUSSION THREAD

**Discussion by acpn** [seleniumbase/SeleniumBase#771] [Q&A] [answered]
Hello,

I'm building a crawler to get some info from social networks and I need to use proxy, here we have a route which give us a proxy dynamically. In README file has an example using `proxy_list.py`, however I need to set the proxy in execution time, something like: when facebook use PROXY_A, when instagram use PROXY_B and so on.

Can I did something like that? If don't, can I use my own **proxy_list** file or set the proxy in self.drive() for example?

Thanks,
Antonio

**Comment by mdmintz** [comment, marked as answer]
@acpn You can set the proxy dynamically by adding the following line in a script:

```python
self.get_new_driver(proxy="SERVER:PORT")
```

That will open a new browser window, and all commands after that will be executed from that window.

**Reply by acpn** [reply]
Awesome @mdmintz exactly what I need, Thanks!!

**Comment by acpn** [comment]
@mdmintz when using firefox has some additional setup that I've to do? I'm asking because when i set `--browser==firefox` I got an error saying when authenticated proxy had to use chrome or edge, makes sense for you?

Thanks,

**Reply by mdmintz** [reply]
Hi @acpn, currently Firefox / Geckodriver does not support an authenticated proxy server. If you want to use an authenticated proxy, you'll need to use Chrome or Edge. You can use a non-authenticated proxy server with Firefox though.
